app/questionnaire/router.py

```diff
@@ -224,22 +224,30 @@ def _get_return_to_location_url(
         if not return_to:
             return None
 
-        if return_to == "calculated-summary" and self.can_access_location(
-            Location(
-                block_id=return_to_block_id,
-                section_id=location.section_id,
-                list_item_id=location.list_item_id,
-            ),
-            routing_path,
-        ):
-            return url_for(
-                "questionnaire.block",
-                block_id=return_to_block_id,
-                list_name=location.list_name,
-                list_item_id=location.list_item_id,
+        if return_to == "grand-calculated-summary" and (
+            url := self._get_return_to_for_grand_calculated_summary(
                 return_to=return_to,
-                _anchor=return_to_answer_id,
+                return_to_block_id=return_to_block_id,
+                location=location,
+                routing_path=routing_path,
+                return_to_answer_id=return_to_answer_id,
+            )
+        ):
+            return url
+
+        if return_to.startswith("calculated-summary") and (
+            url := self._get_return_to_for_calculated_summary(
+                # if there are multiple return to block types, the last one is where to go next
```

Comment:
This is based on the assumption we only support 2 locations right?. Otherwise, it is next in line?.

Reply:
Updated to make it more generic, ie next in line as you say, leaves scope to use it for more items later on